### PR TITLE
Hide confusing/dangerous perms from the OAuth Docs

### PR DESCRIPTION
**Pull request by ProbablePrime**
We have a lot of internal permissions and dangerous permissions that were not flagged as administrative. This fixes that

### CODE REVIEW

File: src/permissions.ts [hunk 1]
```diff
@@ -78,15 +78,19 @@ export const permissions = {
...
+        administrative: true,
     },
     'messages:read:self': {
         text: 'Read your personal messages.',
+        administrative: true,
     },
     'messages:reply:self': {
         text: 'Reply to your personal messages.',
+        administrative: true,
     },
```
Comment (connor4312) [inline src/permissions.ts]: All of these can be deleted, the PM ssystem was removed a while ago

File: src/permissions.ts [hunk 2]
```diff
@@ -106,13 +110,15 @@ export const permissions = {
...
     'resource:delete:self': {
         text: 'Remove your access to emoticons and other graphical resources.',
+        administrative: true,
```
Comment (connor4312) [inline src/permissions.ts]: This should be public

Comment (ProbablePrime) [inline src/permissions.ts]: Can you define what it actually does? The perm suggests "Delete emotes" the text suggests "prevent you from accessing emotes"

Comment (connor4312) [inline src/permissions.ts]: Currently it's not in use anywhere, but it has been in transient use over time for general resource management.

Comment (ProbablePrime) [inline src/permissions.ts]: If it has no use i'd like to keep it hidden from the docs until it has one. At that time we can update its description to be clearer when its function is better defined.

**Pull request merged**